snuba/migrations/table_engines.py

```diff
@@ -0,0 +1,59 @@
+from typing import Mapping, Optional
+
+
+class TableEngine:
```

Comment:
We should have docstrings explaining what these are for.
Which can also carry the warning that these classes need to stay isofunctional when they receive changes so that the result of migrations does not change.